File: DAA/Assignment 1/ass_1.py
# Non-recursive Approach

import time

def fibonacci_iterative(n):
    if n <= 1:
        return n
    a, b = 0, 1
    for _ in range(2, n + 1):
        a, b = b, a + b
    return b

# Example usage
n = 30

start_time = time.time()
result = fibonacci_iterative(n)
end_time = time.time()

execution_time = end_time - start_time
print(f"Iterative Fibonacci({n}) = {result}")
print(f"Execution Time: {execution_time:.6f} seconds")




# Recursive Approach

import time

def fibonacci_recursive(n):
    if n <= 1:
        return n
    return fibonacci_recursive(n - 1) + fibonacci_recursive(n - 2)

# Example usage
n = 30

start_time = time.time()
result = fibonacci_recursive(n)
end_time = time.time()

execution_time = end_time - start_time
print(f"Recursive Fibonacci({n}) = {result}")
print(f"Execution Time: {execution_time:.6f} seconds")

File: DAA/Assignment 1/ass_1.cpp
// Non-recursive Approach

#include <iostream>
#include <chrono>
using namespace std;

int fibonacci_iterative(int n) {
    if (n <= 1)
        return n;
    int a = 0, b = 1;
    for (int i = 2; i <= n; i++) {
        int temp = a + b;
        a = b;
        b = temp;
    }
    return b;
}

int main() {
    int n = 30;

    auto start = chrono::high_resolution_clock::now();
    int result = fibonacci_iterative(n);
    auto stop = chrono::high_resolution_clock::now();

    chrono::duration<double> duration = stop - start;
    
    cout << "Iterative Fibonacci(" << n << ") = " << result << endl;
    cout << "Execution Time: " << duration.count() << " seconds" << endl;

    return 0;
}

// Resursive Approach

/*
// Recursive Approach

#include <iostream>
#include <chrono>
using namespace std;

int fibonacci_recursive(int n) {
    if (n <= 1)
        return n;
    return fibonacci_recursive(n - 1) + fibonacci_recursive(n - 2);
}

int main() {
    int n = 30;

    auto start = chrono::high_resolution_clock::now();
    int result = fibonacci_recursive(n);
    auto stop = chrono::high_resolution_clock::now();

    chrono::duration<double> duration = stop - start;
    
    cout << "Recursive Fibonacci(" << n << ") = " << result << endl;
    cout << "Execution Time: " << duration.count() << " seconds" << endl;

    return 0;
}

*/